src/components/ChatView.tsx

```diff
@@ -410,9 +410,16 @@ const ChatView = forwardRef<ChatViewHandle, ChatViewProps>(({
     };
 
     // Handle save functionality
-    const [showSaveToast, setShowSaveToast] = useState(false);
     const [isSaving, setIsSaving] = useState(false);
 
+    // Unified toast state for all notifications
+    const [showToast, setShowToast] = useState(false);
+    const [toastData, setToastData] = useState({
+        title: '',
+        description: '',
+        emoji: ''
+    });
```

Comment:
why are 2 different variables `showToast` and `toastData` needed? can they not be combined?